streaming/src/main/scala/org/apache/spark/streaming/scheduler/JobGenerator.scala

```diff
@@ -146,6 +146,9 @@ class JobGenerator(jobScheduler: JobScheduler) extends Logging {
         Thread.sleep(pollTime)
       }
       logInfo("Waited for jobs to be processed and checkpoints to be written")
+
+      // do final cleanup in the same thread
+      clearMetadata(Time(clock.getTimeMillis) + graph.getMaxInputStreamRememberDuration)
```

Comment:
`clearMetadata` cannot be called directly here. In addition, it may do another checkpoint which is not needed.